python-lib/sentence_splitter.py

```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+import logging
+from time import perf_counter
+from typing import AnyStr, List
+from spacy.tokens import Doc
+from fastcore.utils import store_attr
+from plugin_io_utils import replace_nan_values, generate_unique
+
+
+class SentenceSplitter:
+    """Module to handle sentence splitting with spaCy for multiple languages
+    Attributes:
+        tokenizer (dict): MultilingualTokenizer instance which stored a dictionary spacy_nlp_dict of spaCy Language instances by language code (key)
+        text_column (str): Name of the dataframe column storing documents to process
+        text_df (pandas DataFrame): DataFrame which contains text_column 
+        language (str) : language of the documents to process. 
+        language_column (str) : Name of the dataframe column storing languages of the document to process. Default to None
+        case_sensitivity (bool): Boolean used to know if the text should be resplitted with lowercase text. 
+    """
+    def __init__(
+        self,
+        text_df,
+        text_column,
+        tokenizer,
+        case_insensitivity,
+        language,
+        language_column=None,
+    ):
+        store_attr()
+
+    def _split_sentences_df(self) -> pd.DataFrame:
+        """Append new column(s) to a dataframe, with documents as lists of sentences
+        Returns: - DataFrame : text_df with the new added column(s) of tokenized text
```

Comment:
Nitpicks: 
- You are missing an element in the return type hint
- can you follow the [google docstring example](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) in terms of line breaks?

Reply:
742bbe5